packages/platform-ios/src/commands/runIOS/index.ts

```diff
@@ -128,9 +128,40 @@ async function runOnSimulator(
     throw new CLIError('Could not parse the simulator list output', error);
   }
 
-  const selectedSimulator = findMatchingSimulator(simulators, args.simulator);
-  if (!selectedSimulator) {
-    throw new CLIError(`Could not find "${args.simulator}" simulator`);
+  let simulator: string = args.simulator;
+  let selectedSimulator: {
+    udid: string;
+    name: string;
+    booted: boolean;
+    version: string;
+  } | null = null;
+
+  /**
+   * If no simulator was chosen, try simulators in following order
+   * - iPhone 11
+   * - iPhone X
+   * - iPhone 8
+   */
+  if (!simulator) {
```

Comment:
I wonder if this could be replaced with a simple `while` loop or a `for` loop and `break` statement. Thoughts?

Reply:
thanks for the suggestion!
I modified it to use `array.some` so that loop would break when simulator is found.

Reply:
That is a good improvement, but still, I find doing side effects inside `some` a bit of a code smell.

How about:
```js
for (const defaultSimulator of defaultSimulators) {
  selectedSimulator = findMatchingSimulator(simulators, defaultSimulator);
  if (selectedSimulator) {
     break;
  }
}
```
or
```js
const selectedSimulator = args.simulator 
   ? findMatchingSimulator(simulators, args.simulator)
   : defaultSimulators.reduce(
       (selectedSimulator, simulator) => selectedSimulator || findMatchingSimulator(simulators, simulator),
       null
     );
```

The below (in my opinion) would be the most concise:
```js
const selectedSimulator = defaultSimulators.reduce(
   (selectedSimulator, simulator) => selectedSimulator || findMatchingSimulator(simulators, simulator),
   findMatchingSimulator(simulators, args.simulator)
);
```
but it would require we bring: `args.simulator` default value to be `iPhone 11` and rename `defaultSimulators` to be `fallbackSimulators` (in other words, try these ones in case user provided/default value does not exist on the machine).

Leaving that up to further discussion.

Reply:
Thank you for the feedback and suggestions!
I find doing side effects inside `some` feels iffy as well, but was not able to come up with a solution at the moment.

I like the last suggestion of providing fallback when `args.simulator` is not available. It let me remove multiple `if` s.

I've modified the default simulator to be `iPhone 11` and modified the document to match the code.